src/context.rs

```diff
@@ -98,6 +101,28 @@ pub trait UsbContext: Clone + Sized + Send + Sync {
         Some(unsafe { DeviceHandle::from_libusb(self.clone(), ptr) })
     }
 
+    /// Opens the device with a pre-opened file descriptor.
+    ///
+    /// This is UNIX-only and platform-specific. It is currently working with
+    /// Linux/Android, but might work with other systems in the future.
+    ///
+    /// This function consumes ownership of the specified file descriptor. The
+    /// device handle will take responsibility for closing it.
```

Comment:
No. This function not consume and device handle not closing it. [libusb_wrap_sys_device](https://libusb.sourceforge.io/api-1.0/group__libusb__dev.html#ga98f783e115ceff4eaf88a60e6439563c)

>The system device handle must remain open until libusb_close() is called. The system device handle will not be closed by libusb_close().

Reply:
Oops! Good catch. I wonder if we shouldn't do it then. What do you think?

Reply:
Hmm... in theory we can create some wrapper struc which take raw handle and implement deref to DeviceHandle. But it's may be not very useful.

Reply:
That's an option, or simply add a Option<RawFd> on DeviceHandle.

Reply:
I dropped that approach, it will be caller responsabiity until we offer some other facilites.